### PR TITLE
NIFI-7633: Added FlowFileConcurrency of SINGLE_BATCH_PER_NODE to allo…

### PR DESCRIPTION
**Pull request by markap14**
…w data to be pulled into a ProcessGroup as a batch

Thank you for submitting a contribution to Apache NiFi.

Please provide a short description of the PR here:

#### Description of PR

_Enables X functionality; fixes bug NIFI-YYYY._

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [ ] Does your PR title start with **NIFI-XXXX** where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?

- [ ] Is your initial contribution a single, squashed commit? _Additional commits in response to PR reviewer feedback should be made on this branch and pushed to allow change tracking. Do not `squash` or use `--force` when pushing to allow for clean monitoring of changes._

### For code changes:
- [ ] Have you ensured that the full suite of tests is executed via `mvn -Pcontrib-check clean install` at the root `nifi` folder?
- [ ] Have you written or updated unit tests to verify your changes?
- [ ] Have you verified that the full build is successful on JDK 8?
- [ ] Have you verified that the full build is successful on JDK 11?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the `LICENSE` file, including the main `LICENSE` file under `nifi-assembly`?
- [ ] If applicable, have you updated the `NOTICE` file, including the main `NOTICE` file found under `nifi-assembly`?
- [ ] If adding new Properties, have you added `.displayName` in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check GitHub Actions CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by pvillard31** [comment]
Will review and play with it during the week...

**Comment by markap14** [comment]
Thanks Pierre. Fixed.

**Comment by pvillard31** [comment]
Hey @markap14 - now that I merged #4345, this PR has conflicts. Can you resolve the conflicts? Happy to review and merge this before 1.12.0 release.

File: nifi-docs/src/main/asciidoc/user-guide.adoc [hunk 1]
```diff
@@ -755,14 +755,19 @@ the parallelization that NiFi uses to process the data. However, there are sever
...
+
 NOTE: The FlowFile Concurrency controls only when data will be pulled into the Process Group from an Input Port. It does not prevent a Processor within the
 Process Group from ingesting data from outside of NiFi.
 
 While the FlowFile Concurrency dictates how data should be brought into the Process Group, the Outbound Policy controls the flow of data out of the Process Group.
 There are two available options for the Outbound Policy: "Stream When Available" and "Batch Output". The default value is "Stream When Available." When this mode is used,
 data that arrives at an Output Port is immediately transferred out of the Process Group, assuming that no backpressure is applied.
 
-The second option is to use "Batch Output." When this Outbound Policy is selected, the Outport Ports will not transfer data out of the Process Group until
+The second option is to use "Batch Output." When this Outbound Policy is selected, the Output Ports will not transfer data out of the Process Group until
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/user-guide.adoc:770]:
```suggestion
The second option is to use "Batch Output". When this Outbound Policy is selected, the Output Ports will not transfer data out of the Process Group until
```

File: nifi-docs/src/main/asciidoc/user-guide.adoc [hunk 2]
```diff
@@ -744,8 +744,8 @@ information or context about the Process Group.
 
 [[Flowfile_Concurrency]]
 === FlowFile Concurrency
-FlowFile Concurrency is used to control how data is brought into the Process Group. There are two options available: Unbounded (which is the default)
-and Single FlowFile Per Node. When the concurrency is set to "Unbounded," the Input Ports in the Process Group will ingest data as quickly as they
+FlowFile Concurrency is used to control how data is brought into the Process Group. There are three options available: Unbounded (which is the default),
+Single FlowFile Per Node, and Single Batch Per Node. When the concurrency is set to "Unbounded," the Input Ports in the Process Group will ingest data as quickly as they
 are able, provided that backpressure does not prevent them from doing so.
 
 When the FlowFile Concurrency is configured to "Single FlowFile Per Node," the Input Ports will only allow through a single FlowFile at at time.
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/user-guide.adoc:751]:
```suggestion
When the FlowFile Concurrency is configured to "Single FlowFile Per Node", the Input Ports will only allow through a single FlowFile at at time.
```

File: nifi-docs/src/main/asciidoc/user-guide.adoc [hunk 3]
```diff
@@ -755,14 +755,19 @@ the parallelization that NiFi uses to process the data. However, there are sever
 is one in which each incoming FlowFile contains references to several other data items, such as a list of files in a directory. The user may want to
 process the entire listing before allowing any other data to enter the Process Group.
 
+When the FlowFile Concurrency is configured to "Single Batch Per Node," the Input Ports will behave similarly to the way that they behave in the
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/user-guide.adoc:758]:
```suggestion
When the FlowFile Concurrency is configured to "Single Batch Per Node", the Input Ports will behave similarly to the way that they behave in the
```

File: nifi-docs/src/main/asciidoc/user-guide.adoc [hunk 4]
```diff
@@ -755,14 +755,19 @@ the parallelization that NiFi uses to process the data. However, there are sever
...
+"Single FlowFile Per Node" mode, but when a FlowFile is ingested, the Input Ports will continue to ingest all data until all of the queues feeding
+the Input Ports have been emptied. At that point, they will not bring any more data into the Process Group until all data has finished processing and
+has left the Process Group (see note on <<Connecting_Batch_Oriented_Groups>> below).
+
 NOTE: The FlowFile Concurrency controls only when data will be pulled into the Process Group from an Input Port. It does not prevent a Processor within the
 Process Group from ingesting data from outside of NiFi.
 
 While the FlowFile Concurrency dictates how data should be brought into the Process Group, the Outbound Policy controls the flow of data out of the Process Group.
 There are two available options for the Outbound Policy: "Stream When Available" and "Batch Output". The default value is "Stream When Available." When this mode is used,
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/user-guide.adoc:767]:
```suggestion
There are two available options for the Outbound Policy: "Stream When Available" and "Batch Output". The default value is "Stream When Available". When this mode is used,
```

File: nifi-docs/src/main/asciidoc/user-guide.adoc [hunk 5]
```diff
@@ -778,6 +783,25 @@ batch.output.B = 1, batch.output.C = 2, batch.output.D = 0.
 The Outbound Policy of "Batch Output" doesn't provide any benefits when used in conjunction with a FlowFile Concurrency of "Unbounded."
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/user-guide.adoc:783]:
```suggestion
The Outbound Policy of "Batch Output" doesn't provide any benefits when used in conjunction with a FlowFile Concurrency of "Unbounded".
```

File: nifi-docs/src/main/asciidoc/user-guide.adoc [hunk 6]
```diff
@@ -778,6 +783,25 @@ batch.output.B = 1, batch.output.C = 2, batch.output.D = 0.
...
 As a result, the Outbound Policy is ignored if the FlowFile Concurrency is set to "Unbounded."
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/user-guide.adoc:784]:
```suggestion
As a result, the Outbound Policy is ignored if the FlowFile Concurrency is set to "Unbounded".
```

**Comment by pvillard31** [comment]
Merged to main, thanks @markap14 !

**Pull request closed without merging**